app/api/helpers/third_party_auth.py

```diff
@@ -59,8 +59,8 @@ def get_client_secret(cls):
 
     @classmethod
     def get_redirect_uri(cls):
-        url = urllib.parse.urlparse(request.url)
-        fb_redirect_uri = 'http://localhost:4200/oauth/callback?provider=facebook'
+        url = get_settings()['frontend_url'] + '/oauth/callback?provider=facebook'
```

Comment:
There's a util method called get_frontend_url. Use that. Also, remove WIP from title if you're done and want to get the PR reviewed. Mentors ignore reviewing PRs with WIP in title